### ISSUE TITLE
PANIC error on upsert query

### ISSUE THREAD

**Issue by hugo082** [prisma/prisma-client-js#683]
## Bug description

Prisma Client JS panic on upsert query.

```typescript
const b = await prisma.enitityB.create({
  data: {
    name: "B NAME"
  }
})

// ==== Panic here ====
const a = await prisma.enititA.upsert({
  where: {
    connectorId_indicator_query: {
      connectorId: b.id,
      indicator: "INDICATOR_A",
      query: "QUERY_A",
    }
  },
  create: {
    connector: { connect: { id: b.id } },
    indicator: "INDICATOR_A",
    query: "QUERY_A",
    name: "A NAME"
  },
  update: {
    name: "A NAME updated"
  }
})
```

Prisma panic error:
```bash
PANIC: called `Option::unwrap()` on a `None` value
    at PrismaClientFetcher.request (/private/tmp/prisma-panic/node_modules/@prisma/client/src/runtime/getPrismaClient.ts:643:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
Error in Prisma Client:
PANIC: called `Option::unwrap()` on a `None` value in
query-engine/core/src/query_graph_builder/extractors/filters.rs:279:35
```

## How to reproduce

You can reproduce with this repository: https://github.com/hugo082/prisma-client-issue

## Expected behavior

The entity A must be created and prisma client must not panic.

## Prisma information

Prisma schema:
 
```prisma

model EnititA {
  id          String    @default(cuid()) @id
  name        String?
  connector   EnitityB? @relation(fields: [connectorId], references: [id])
  connectorId String?   @map("connector_id")
  indicator   String
  query       String

  @@unique([connectorId, indicator, query])
}

model EnitityB {
  id      String    @default(cuid()) @id
  name    String?
  EnititA EnititA[]
}
```

## Environment & setup

```bash
@prisma/cli          : 2.0.0-beta.4
Current platform     : darwin
Query Engine         : query-engine afd294205618b1c825b013ba6f5a6ebe4aa4a514 (at ***/yarn/global/node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli afd294205618b1c825b013ba6f5a6ebe4aa4a514 (at ***/yarn/global/node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core afd294205618b1c825b013ba6f5a6ebe4aa4a514 (at ***/yarn/global/node_modules/@prisma/cli/introspection-engine-darwin)
```

**Comment by pantharshit00** [comment]
Looks related to https://github.com/prisma/prisma/issues/2353 and https://github.com/prisma/prisma/issues/2353

I can confirm this though:
![image](https://user-images.githubusercontent.com/22195362/82350606-327af400-9a19-11ea-8ac1-99a970cbd806.png)

 Version used for reproduction:
```
Environment variables loaded from ./prisma/.env
@prisma/cli          : 2.0.0-alpha.1228
Current platform     : debian-openssl-1.1.x
Query Engine         : query-engine 43c2992b55362e1807325b149fa5132d7bd4751f (at /home/harshit/code/reproductions/issue-client-683/prisma-client-issue/node_modules/@prisma/cli/query-engine-debian-openssl-1.1.x)
Migration Engine     : migration-engine-cli 43c2992b55362e1807325b149fa5132d7bd4751f (at /home/harshit/code/reproductions/issue-client-683/prisma-client-issue/node_modules/@prisma/cli/migration-engine-debian-openssl-1.1.x)
Introspection Engine : introspection-core 43c2992b55362e1807325b149fa5132d7bd4751f (at /home/harshit/code/reproductions/issue-client-683/prisma-client-issue/node_modules/@prisma/cli/introspection-engine-debian-openssl-1.1.x)
Format Binary        : prisma-fmt 43c2992b55362e1807325b149fa5132d7bd4751f (at /home/harshit/code/reproductions/issue-client-683/prisma-client-issue/node_modules/@prisma/cli/prisma-fmt-debian-openssl-1.1.x)
```

**Comment by dpetrick** [comment]
Fixed from 2.0.0-alpha.1230 onwards.
